libs/langgraph/tests/test_pregel.py

```diff
@@ -11027,3 +11027,46 @@ def _node(state: State):
     app = parent.compile()
 
     assert app.get_graph(xray=True).draw_mermaid() == snapshot
+
+
+def test_subgraph_retries():
+    class State(TypedDict):
+        count: int
+
+    class ChildState(State):
+        some_list: Annotated[list, operator.add]
+
+    class RandomError(ValueError):
+        """This will be retried on."""
+
+    def parent_node(state: State):
+        return {"count": state["count"] + 1}
+
+    def child_node_a(state: ChildState):
+        assert not state.get(
+            "some_list"
+        ), f"State shouldn't accomulate on retries. Got: {state['some_list']}"
+        return {"some_list": ["val"]}
+
+    def child_node_b(state: ChildState):
+        if state["count"] == 1:
+            raise RandomError("First attempt fails")
+        return {"count": state["count"] + 1}
+
+    child = StateGraph(ChildState)
+    child.add_node(child_node_a)
+    child.add_node(child_node_b)
+    child.add_edge("__start__", "child_node_a")
+    child.add_edge("child_node_a", "child_node_b")
+
+    parent = StateGraph(State)
+    parent.add_node("parent_node", parent_node)
+    parent.add_node("child_graph", child.compile())
+    parent.add_edge("parent_node", "child_graph")
+    parent.set_entry_point("parent_node")
+
+    checkpointer = MemorySaver()
+    app = parent.compile(checkpointer=checkpointer)
+    app.retry_policy = RetryPolicy(max_attempts=3, retry_on=(RandomError,))
```

Comment:
btw the "canonical" way of adding retry policy is in add_node

Reply:
oh got it

Reply:
hadn't realized that